src/deploy/functions/build.ts

```diff
@@ -224,12 +224,39 @@ export async function resolveBackend(
   const projectId = userEnvOpt.projectId;
   let paramValues: Record<string, Field<string | number | boolean>> = {};
   if (previews.functionsparams) {
-    paramValues = await params.resolveParams(build.params, projectId, userEnvs);
+    paramValues = await params.resolveParams(build.params, projectId, envWithTypes(userEnvs));
+
+    // TODO(vsfan@): when merging secrets support into the Build, make sure we aren't writing those to disk.
+    const toWrite: Record<string, string> = {};
+    for (const paramName of Object.keys(paramValues)) {
+      if (userEnvs.hasOwnProperty(paramName)) {
+        continue;
+      }
+      toWrite[paramName] = paramValues[paramName]?.toString() || "";
+    }
+    writeUserEnvs(toWrite, userEnvOpt);
   }
 
   return toBackend(build, paramValues);
 }
 
+function envWithTypes(rawEnvs: Record<string, string>): Record<string, string | number | boolean> {
+  const out: Record<string, string | number | boolean> = {};
+  for (const envName of Object.keys(rawEnvs)) {
+    const value = rawEnvs[envName];
+    if (!isNaN(+value) && isFinite(+value) && !value.includes("e")) {
```

Comment:
I assume this is intended to support ints instead of floats (I thought we supported both parameter types). Technically "1e3" is an int but would fail this test. I think you might just want to check for +value being betwee than `Number.MIN_SAFE_INTEGER` and `Number.MAX_SAFE_INTEGER`

Reply:
The hilarious thing is that the reason it's like this in the first place is that my first swing at this _did_ allow 1e3 and daniel had some reservations about it--I've come around to his position personally because we expect to support params in multiple languages and in most languages that aren't javascript, I don't think you can write int literals like that.